### PR TITLE
Rename WGPUDeviceLostReason_Undefined to Unknown

### LINKED ISSUE #223: rename WGPUDeviceLostReason "Undefined" to "Unknown"
This was changed in the JS API: https://github.com/gpuweb/gpuweb/pull/3976

### PR DESCRIPTION
**Pull request by hocheung-chromium**
Please check
https://issues.chromium.org/issues/42241081 for
more information and progress.

@kainino0x @austinEng 

Fixes #223

### CODE REVIEW

**Comment by austinEng** [comment]
I believe WGPUDeviceLostReason_Unknown should start with value 1, not 0

**Comment by hocheung-chromium** [comment]
> I believe WGPUDeviceLostReason_Unknown should start with value 1, not 0

@austinEng 

> I believe WGPUDeviceLostReason_Unknown should start with value 1, not 0

I have redefined the values ​​for `Unknown` and `Destoryed` in the yml file

@austinEng

**Review by austinEng — APPROVED** [review]

**Comment by hocheung-chromium** [comment]
Hello Kai Ninomiya, could you please take a moment to review this PR? Thanks.

@kainino0x

**Comment by hocheung-chromium** [comment]
I also followed your instructions and kept `Undefined` here. Please review, thank you.

@kainino0x

File: webgpu.yml [hunk 1]
```diff
@@ -436,9 +436,15 @@ enums:
       TODO
     entries:
       - name: undefined
+        value: 0x0001
+        doc: |
+          TODO
```
Comment (kainino0x) [inline webgpu.yml:441]: We don't need `undefined` here; we only have it in Dawn as an intermediate deprecation before we remove it. In this repo we are just trying to converge to 1.0. So we can skip the deprecation.
```suggestion
```

File: webgpu.h [hunk 2]
```diff
@@ -292,8 +292,9 @@ typedef enum WGPUCullMode {
 } WGPUCullMode WGPU_ENUM_ATTRIBUTE;
 
 typedef enum WGPUDeviceLostReason {
-    WGPUDeviceLostReason_Undefined = 0x00000000,
-    WGPUDeviceLostReason_Destroyed = 0x00000001,
+    WGPUDeviceLostReason_Undefined = 0x00000001,
```
Comment (kainino0x) [inline webgpu.h:295]:
```suggestion
```

**Review by kainino0x — COMMENTED** [review]
I'll link #223 from the summary.

**Review by kainino0x — APPROVED** [review]

**Comment by kainino0x** [comment]
> I also followed your instructions and kept `Undefined` here. Please review, thank you.

As mentioned we don't actually need it for this repository. Sorry the state of things is confusing right now. Also apologies for the review delay!

I removed `Undefined`, so LGTM!

**Comment by gz83** [comment]
> > I also followed your instructions and kept `Undefined` here. Please review, thank you.
> 
> As mentioned we don't actually need it for this repository. Sorry the state of things is confusing right now. Also apologies for the review delay!
> 
> I removed `Undefined`, so LGTM!

I see what you mean, thank you very much for reviewing my PR.

@kainino0x

**Review by rajveermalviya — APPROVED** [review]

**Pull request merged**